docs/csharp/fundamentals/types/classes.md

```diff
@@ -20,75 +20,86 @@ MyClass mc = new MyClass();
+
+:::code source="./snippets/classes/Containers.cs" id="ContainerPrimaryConstructor":::
+
+Adding parameters to the class name defines the *primary constructor*. Those parameters are available in the class body. You can use the them to initialize fields, or anywhere else in members where you need that constructor parameter.
+
+You can also use the `required` modifier on a property and allow callers to use an *object initializer* to set the initial value of a property:
+
+:::code source="./snippets/classes/Program.cs" id="RequiredProperties":::
+
+The addition of the `required` keyword mandates that callers must set those properties as part of a `new` expression:
+
+```csharp
+var p1 = new Person(); // Error! Required properties not set
+var p2 = new Person() {FirstName = "Grace", LastName = "Hopper" };
 ```
-  
- This code creates two object references that both refer to the same object. Therefore, any changes to the object made through `object3` are reflected in subsequent uses of `object4`. Because objects that are based on classes are referred to by reference, classes are known as reference types.  
-  
-## Class inheritance  
 
-Classes fully support *inheritance*, a fundamental characteristic of object-oriented programming. When you create a class, you can inherit from any other class that is not defined as [`sealed`](../../language-reference/keywords/sealed.md), and other classes can inherit from your class and override class virtual methods. Furthermore, you can implement one or more interfaces.
+## Class inheritance
+
+Classes fully support *inheritance*, a fundamental characteristic of object-oriented programming. When you create a class, you can inherit from any other class that isn't defined as [`sealed`](../../language-reference/keywords/sealed.md), and other classes can inherit from your class and override class virtual methods. Furthermore, you can implement one or more interfaces.
 
 Inheritance is accomplished by using a *derivation*, which means a class is declared by using a *base class* from which it inherits data and behavior. A base class is specified by appending a colon and the name of the base class following the derived class name, like this:
 
- ```csharp
- public class Manager : Employee
- {
-     // Employee fields, properties, methods and events are inherited
-     // New Manager fields, properties, methods and events go here...
- }
- ```
+:::code source="./snippets/classes/Program.cs" id="DerivedClass":::
 
 When a class declares a base class, it inherits all the members of the base class except the constructors. For more information, see [Inheritance](../object-oriented/inheritance.md).
 
 A class in C# can only directly inherit from one base class. However, because a base class may itself inherit from another class, a class may indirectly inherit multiple base classes. Furthermore, a class can directly implement one or more interfaces. For more information, see [Interfaces](interfaces.md).
```

Comment:
```suggestion
A class in C# can only directly inherit from one base class. However, because a base class may itself inherit from another class, a class might indirectly inherit multiple base classes. Furthermore, a class can directly implement one or more interfaces. For more information, see [Interfaces](interfaces.md).
```